src/api/middleware/company.js

```diff
@@ -110,15 +110,28 @@ export const isNotDisabled = (owner) => async (req, res, next) => {
 };
 
 export const canManageAccountSettings = (companyId) => async (req, res, next) => {
-    const company = await (new CompanyService()).findById(companyId, true);
-
-    // only god or the same company can change account settings
-    if (!req.hasAdminPrivileges && company._id.toString() !== req.targetOwner) {
-        return next(new APIError(
-            HTTPStatus.FORBIDDEN,
-            ErrorTypes.FORBIDDEN,
-            ValidationReasons.INSUFFICIENT_PERMISSIONS_COMPANY_SETTINGS
-        ));
+    try {
+        const company = await (new CompanyService()).findById(companyId, true);
+
+        // company not found
+        if (!company) {
+            return next(new APIError(
+                HTTPStatus.NOT_FOUND,
+                ErrorTypes.VALIDATION_ERROR,
+                ValidationReasons.COMPANY_NOT_FOUND
+            ));
+        }
```

Comment:
Why do we add this in `canManageAccountSettings`? Other endpoints use the validator for this (might change in #309 ) so we should be consistent

Reply:
There was indeed no need for this, I removed it.